### ISSUE TITLE
ttnn.fmod unary low PCC when scalar is between -0.003 and 0.003

### ISSUE THREAD

**Issue by amalbasaTT** [tenstorrent/tt-metal#17362]
**To Reproduce**
Run the unit test bellow:
```
# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.

# SPDX-License-Identifier: Apache-2.0

from loguru import logger
import random
import pytest
import torch
import ttnn
import traceback

from tests.ttnn.utils_for_testing import check_with_pcc
from tests.sweep_framework.sweep_utils.utils import tensor_to_dtype
from models.utility_functions import skip_for_grayskull


def run_fmod_unary_tests(
    input_shape,
    scalar,
    dtype,
    dlayout,
    in_mem_cfg,
    out_mem_cfg,
    device,
):
    torch.manual_seed(0)
    
    x = tensor_to_dtype(torch.Tensor(size=input_shape).uniform_(-100, 100), dtype)

    try:
        # get ref result
        golden_function = ttnn.get_golden_function(ttnn.fmod)
        ref_value = golden_function(x, scalar)
        
        tt_input = ttnn.from_torch(x, dtype=dtype, layout=dlayout, device=device, memory_config=in_mem_cfg)

        tt_result = ttnn.fmod(tt_input, scalar, memory_config=out_mem_cfg)
        tt_result = ttnn.to_torch(tt_result)

    except Exception as e:
        logger.warning(f"Test execution crashed: {e}")
        print(traceback.format_exc())
        raise e

    pcc, message = check_with_pcc(ref_value, tt_result, 0.999)
    assert pcc, message


test_sweep_args = [
    (
        (6, 5, 90, 112),
        torch.tensor(1, dtype=torch.bfloat16).uniform_(-0.003, 0.003).item(),
        ttnn.bfloat16,
        ttnn.TILE_LAYOUT,
        ttnn.DRAM_MEMORY_CONFIG,
        ttnn.DRAM_MEMORY_CONFIG,
    ),
    (
        (2, 1, 120, 11),
        torch.tensor(1, dtype=torch.bfloat16).uniform_(-0.003, 0.003).item(),
        ttnn.bfloat16,
        ttnn.TILE_LAYOUT,
        ttnn.L1_MEMORY_CONFIG,
        ttnn.DRAM_MEMORY_CONFIG,
    ),
    (
        (3, 123, 115),
        torch.tensor(1, dtype=torch.bfloat16).uniform_(-0.003, 0.003).item(),
        ttnn.bfloat16,
        ttnn.TILE_LAYOUT,
        ttnn.DRAM_MEMORY_CONFIG,
        ttnn.L1_MEMORY_CONFIG,
    ),
    (
        (69, 178),
        torch.tensor(1, dtype=torch.bfloat16).uniform_(-0.003, 0.003).item(),
        ttnn.bfloat16,
        ttnn.TILE_LAYOUT,
        ttnn.L1_MEMORY_CONFIG,
        ttnn.L1_MEMORY_CONFIG,
    ),
]


@pytest.mark.parametrize(
    "input_shape, scalar, dtype, dlayout, in_mem_config, out_mem_config",
    (test_sweep_args),
)
@skip_for_grayskull("Not supported for Grayskull")
def test_fmod_unary(input_shape, scalar, dtype, dlayout, in_mem_config, out_mem_config, device):
    run_fmod_unary_tests(input_shape, scalar, dtype, dlayout, in_mem_config, out_mem_config, device)

```

**Comment by mouliraj-mcw** [comment]
same as #17361 .
[Look here](https://github.com/tenstorrent/tt-metal/issues/17361#issuecomment-2649957633)

**Comment by mcw-anasuya** [comment]
Please find the obserations [here](https://github.com/tenstorrent/tt-metal/issues/17361#issuecomment-2748790569).
PR increasing the overall accuracy has been merged to main. Hence closing.
